src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -338,17 +338,34 @@ private async Task RecoverStoreTipAsync(ChainedHeader chainedHeader)
                 throw new ProvenBlockHeaderException("Proven block header store failed to recover.");
             }
 
-            if (tipHeight > chainedHeader.Height)
+            ChainedHeader pendingTip = null;
+
+            // This happens when the new chain reorg is behind this current chain store.
+            if (newChainedHeader.Height <= tipHeight)
```

Comment:
The condition is only when `inedHeader.Height < tipHeight`
then you need an else `if(inedHeader.Height > tipHeight` you need to throw.